docs/enhancements/patch-install-manifests.md

```diff
@@ -0,0 +1,137 @@
+
+## API
+
+### `ClusterDeploymentCustomization.Spec.InstallerManifestPatches`
+
+**Example:**
+```yaml
+apiVersion: v1
+kind: ClusterDeploymentCustomization
+metadata:
+  name: foo-cluster-deployment-customization
+  namespace: my-project
+spec:
+  installerManifestPatches:
+    # Add custom labels to all master machine manifests
+    - manifestSelector:
+        glob: openshift/99_openshift-cluster-api_master-machines-*.yaml
+      patches:
+        - op: add
+          path: metadata/labels/a-custom-label
+          value: foo
+        - op: add
+          path: metadata/labels/b-custom-label
+          value: foo
+    - ...
+  ...
+```
+
+- Build on the existing [groundwork](#groundwork), extending `ClusterDeploymentCustomization` to
+  expose a new field, `InstallerManifestPatches`.
+- Because installer generates multiple manifests, and RFC 6902 syntax can only reference a single
+  document at a time, we can't directly use `PatchEntity`; we must use an intervening type to
+  identify which file(s) a patch is to be applied to.
+  We'll call this `InstallerManifestPatch`.
+- Each `InstallerManifestPatch`:
+  - Identifies one or more files via subfield `ManifestSelector`.
+  - Lists `PatchEntity`s to apply to files thus identified.
+    This is the same `PatchEntity` used for `InstallConfigPatches`.
+- `ManifestSelector` supports one subfield, `Glob`, which accepts a path matching string as
+  supported by golang's [filepath.Glob](https://pkg.go.dev/path/filepath#Glob).
+  - We will execute the glob relative to the working directory in which manifests were generated.
+  - For security, we will error on any glob starting with `/` or containing `../`.
+- `ManifestSelector` is extensible.
+  E.g. in the future we may wish to match manifests based on the GVK of the object therein.
+
+### `ClusterDeployment.Spec.Provisioning.CustomizationRef`
+
+**Example:**
+```yaml
+apiVersion: v1
+kind: ClusterDeployment
+metadata:
+  name: foo-cluster-deployment
+  namespace: my-project
+spec:
+  provisioning:
+    installConfigSecretRef: ic-secret
+    customizationRef: foo-cluster-deployment-customization
+    ...
+  ...
+```
+
+- Extend `ClusterDeployment`, adding a `CustomizationRef` field at the same level as
+  `InstallConfigSecretRef`, i.e. under `Spec.Provisioning`.
+- `CustomizationRef` is the name of a `ClusterDeploymentCustomization` in the same namespace as the
+  `ClusterDeployment`.
+  We will attempt to load it and act on each entry.
+
+### Status
+**TBD.**
+
+Where/how do we report the success/failure of
+- CDC as a whole? In a new `CD.Status.Conditions[?Type=CustomizationsSuccessful]` or similar?
+- Individial `CDC.InstallerManifestPatches`?
+  How are we doing it for `InstallConfigPatches` for ClusterPools?
+
+## Failure Modes
+**TBD.**
+
+- Is it a failure if `ManifestSelector` matches zero manifests?
+- Is it a failure if a `Patches[].Path` matches no paths?
+- Failure to execute a patch should probably bubble up as an overall failure.
```

Comment:
In case of InstallConfigPatches, the CP resolution fails and the failure is saved in the CDC which causes CP to pick a different CDC in the next attempt.